### PR TITLE
Unignore and fix doctest

### LINKED ISSUE #849: TST: Do not skip valid doctest examples
For example, I see you download real data in `docs/spectral_cube.rst` but then you skip testing actually using the object generated by the data.

```python
>>> sc.shape #doctest:+SKIP
(74, 74, 4563)
```

Why not just run the whole example with doctest?

### PR DESCRIPTION
**Pull request by pllim**
This PR is to address this issue:

* fix #849 

Also remove the doctest directive that used to ignore output. I actually uncovered some wrong code used in examples when I am doing this. Those are fixed here too.

I came across #1016 but that is a battle for another day.

### CODE REVIEW

File: specutils/manipulation/model_replace.py [hunk 1]
```diff
@@ -64,7 +64,8 @@ def model_replace(spectrum, replace_region, model=10, extrapolation_treatment='d
     >>> flux_val = np.array([2, 4, 6, 8, 10, 12, 14, 16, 18, 20])
     >>> input_spectrum = Spectrum1D(spectral_axis=wave_val * u.AA, flux=flux_val * u.mJy)
     >>> spline_knots = [3.5, 4.7, 6.8, 7.1] * u.AA
-    >>> result = model_replace(input_spectrum, None, spline_knots) # doctest: +IGNORE_OUTPUT
+    >>> model_replace(input_spectrum, None, spline_knots)
+    <Spectrum1D(flux=<Quantity [ 2.,  4.,  6.,  8., 10., 12., 14., 16., 18., 20.] mJy>, spectral_axis=<SpectralAxis [ 1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.,  9., 10.] Angstrom>)>
```
Comment (pllim) [inline specutils/manipulation/model_replace.py:68]: I dunno what this example is supposed to return. I don't have anything to refer to but the output is same as input, is that correct?

Comment (rosteen) [reply specutils/manipulation/model_replace.py:68]: The input spectrum is just a straight line, so I think a spline fit should just return a straight line as well. Makes sense to me, although it isn't a particularly enlightening example.

Comment (pllim) [reply specutils/manipulation/model_replace.py:68]: Thanks for the clarification!

I think making examples enlightening is out of scope here. 😆

**Review by rosteen — APPROVED** [review]
Looks good to me, docs pass and a quick look at the rendered docs looks good. Thanks!

**Comment by pllim** [comment]
Thanks for the review!

**Pull request merged**